src/basic/log.c

```diff
@@ -1341,6 +1344,14 @@ void log_received_signal(int level, const struct signalfd_siginfo *si) {
                          signal_to_string(si->ssi_signo));
 }
 
+void set_log_syntax_callback(log_syntax_callback_t cb, void *userdata) {
+        assert(cb || log_syntax_callback);
+        assert(userdata || log_syntax_callback_userdata);
```

Comment:
i'd not verify userdata like this. We generally don't make the requirement that userdata should be non-NULL. consumers might not need the userdata pointer and will pass NUL, and that is OK. 

maybe replace this line with:

```c
assert(cb || !userdata);
```

i.e. it's not OK to pass non-NULL userdata without also passing a callback